### DISCUSSION TITLE
Tweak widgets on the fly

### DISCUSSION THREAD

**Discussion by mcnuggetsx20** [qtile/qtile#3883] [Q&A]
Is it possible to change parameters of a certain widget (e.g the background color) on the fly? I was thinking maybe one could do it by accessing the `qtile.widgets_map`, like it happens with `update()` function for a TextBox.

**Comment by elParaguayo** [comment]
Yes. You can.
You could do something like:
``` python
@lazy.widget["widgetname"].function
def change_background(widget):
    widget.background = "ff0000"
    widget.draw()
```
Then bind `change_background()` to a key.
